### ISSUE TITLE
Filter By Price Widget on the Front Page not working

### ISSUE THREAD

**Issue by felipemarcos** [woothemes/woocommerce#35768]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Filter by Price Widget does not work if the Shop page is also the Front Page.

Going to `https://yoursite.com` loads the front page as expected.

Going to `https://yoursite.com/?min_price=0&max_price=90` instead of showing the Shop page, it acts like the Blog page and loads the Post Archive template instead.

**Related**
https://core.trac.wordpress.org/ticket/25143
https://wordpress.stackexchange.com/questions/20946/as-soon-as-i-add-a-custom-query-var-using-the-filter-my-pages-point-to-an-archi

### Expected behavior

It should load the frontpage even if the query vars `min_price` or `max_price` are set in the URL.

### Actual behavior

It loads the Post Archive template instead.

### Steps to reproduce

1. Go to Settings > Reading > Homepage - Set it to "Shop" and save
2. Appearance > Widgets > Add the "Filter by Price" block to the Sidebar and save
3. Go to the Frontpage and try to filter by price

Screen recording: https://www.loom.com/share/199d7bd894f14b2abaf48d1b815bd536

### WordPress Environment

<details>
 <summary>View log</summary>

`
### WordPress Environment ###

WordPress address (URL): http://woo.local
Site address (URL): http://woo.local
WC Version: 7.1.0
REST API Version: ✔ 7.1.0
WC Blocks Version: ✔ 8.7.5
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.1.1
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 8.1.9
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 7.84.0
(SecureTransport) OpenSSL/1.1.1q

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 7.1.0
WC Database Prefix: wp_
Total Database Size: 5.03MB
Database Data Size: 3.54MB
Database Index Size: 1.49MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 2.52MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.11MB + Index: 0.06MB + Engine InnoDB
wp_posts: Data: 0.05MB + Index: 0.06MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 24
page: 7
post: 2
product: 18
product_variation: 7

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 7.1.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: BRL (R$)
Currency Position: left_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔

### WC Pages ###

Shop base: #5 - /
Cart: #6 - /cart/
Checkout: #7 - /checkout/
My account: #8 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 4.2.0
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
multichannel-marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
settings

Daily Cron: ✔ Next scheduled: 2022-11-30 18:49:09 +00:00
Options: ✔
Notes: 36
Onboarding: completed

### Action Scheduler ###

Complete: 35
Oldest: 2022-11-29 18:51:15 +0000
Newest: 2022-11-29 18:51:15 +0000

Pending: 1
Oldest: 2022-11-30 18:49:11 +0000
Newest: 2022-11-30 18:49:11 +0000

### Status report information ###

Generated at: 2022-11-29 18:56:14 +00:00
`
</details>

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by felipemarcos** [comment]
These query vars are set in `wp-content/plugins/woocommerce/packages/woocommerce-blocks/src/BlockTypes/ProductQuery.php` on line 144

**Comment by swatipawarGS** [comment]
Hi @FelipeMarcos,

Thank you for taking the time to report this bug, we really appreciate your help. We are able to reproduce the issue on our end using the  WooCommerce 7.1.0 version and WordPress 6.1.1 version. 

Please find below the screencast for reference :

https://user-images.githubusercontent.com/94531721/204986362-d72f5ea4-e1a5-4625-8a03-aa3de8831e13.mp4

Priority of this issue will be assigned shortly after we’d go through a round of prioritization. Please note it may take a couple of days.

**Comment by bitmakers** [comment]
After today's update (7.2.0) this is now also happening with the "Filter Products by Attribute" widget - please fix this ASAP, it breaks a ton of sites.

**Comment by felipemarcos** [comment]
> Priority of this issue will be assigned shortly after we’d go through a round of prioritization. Please note it may take a couple of days.

Hi @swatipawarGS. Would you happen to know when the issue will be assigned? Thanks!

**Comment by peterfabian** [comment]
cc @nerrad for visibility since the issue was originally reported for blocks, but now it seems it's also affecting classic widgets, so not sure if it makes sense to transfer it.

**Comment by nerrad** [comment]
Hard to tell if this is actually classic widgets or if it's the filter blocks being used in the block widget editor. I'm leaning more towards the latter because it sounds like something that might happen with the blocks not accounting for the routing correctly.

cc @danielwrobert and @Aljullu can either of you try to reproduce this (or get this week's porter to) with the Filter by Price or Attribute blocks following the original poster's instructions and if reproducible move it to the Woo Blocks repo. If it can be reproduced this sounds like something that should be prioritized.

**Comment by Aljullu** [comment]
> Hard to tell if this is actually classic widgets or if it's the filter blocks being used in the block widget editor. I'm leaning more towards the latter because it sounds like something that might happen with the blocks not accounting for the routing correctly.

According to the videos shared, it seems to be the block. However, I'm unable to reproduce. I tested on three different sites (local and online) and there was no luck, it's working as expected in all cases for me:

[Enregistrament de pantalla des de 2023-05-22 16-38-12.webm](https://github.com/woocommerce/woocommerce/assets/3616980/1e925b7f-ee8d-40fe-9554-10e194a3a793)

@danielwrobert can you give it a try? Maybe this was fixed in a recent release of WP or WC? In any case, I think we can safely transfer this issue to the WC Blocks repo.

**Comment by danielwrobert** [comment]
I am also unable to reproduce this issue. I've tested with the following conditions:

1. Storefront theme – both with and without the Woo Blocks plugin enabled.
2. Twenty Twenty-One theme – both with and without the Woo Blocks plugin enabled.

Everything seems to filter as expected in all of the above scenarios.

> This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).

_Note that I only tested with Twenty Twenty-One, since **Appearance > Widgets** is no longer available in TT2 and up._
